[testing] Running unpackaged not working on net10.0

### Description

We made some refactoring to allow to build and run be separate steps and to run some tests for Android on linux on pr https://github.com/dotnet/maui/pull/28065 . 

When merging these changes to net10.0 branch a problem seems to appear running the Windows unpackaged apps device tests. The run just hangs for 4 hours because it doesn't find the correct runtime. 

https://dev.azure.com/xamarin/public/_build/results?buildId=137711&view=results

<img width="454" alt="Image" src="https://github.com/user-attachments/assets/68f0013c-1216-43f8-9525-75fce8484692" />

I have [disable these tests ](https://github.com/dotnet/maui/pull/28346/commits/e6ea5ccf06f5fc377138ac4a3bba524b98a995e5)for now on net10.0 branch till we figure why it fails and how to fix it . I think the problem is that net10 doesn't t exist on the machine. 

Previous run that worked ok :

https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=11178507&view=logs&j=dc19ef30-62b7-5706-c779-8e65a8085566

### Steps to Reproduce

Enable the unpackaged tests on Device tests yaml and try to run them 

```
  windowsVersions: ['packaged', 'unpackaged']
```

### Affected platforms

Windows

### Relevant log output

```shell
testOnly
========================================
Executing task: testOnly
Cleaning directory D:/a/1/a/test-results
Cleaned directories
Test Results File: D:\a\1\a\test-results\TestResults-com_microsoft_maui_essentials_devicetests.xml
Tests To Run File: D:\a\1\a\test-results\devicetestcategories.txt
Executing: "D:/a/1/s/artifacts/bin/Essentials.DeviceTests/Debug/net10.0-windows10.0.19041.0/win-x64/Microsoft.Maui.Essentials.DeviceTests.exe" D:\a\1\a\test-results\TestResults-com_microsoft_maui_essentials_devicetests.xml
You must install or update .NET to run this application.

App: D:\a\1\s\artifacts\bin\Essentials.DeviceTests\Debug\net10.0-windows10.0.19041.0\win-x64\Microsoft.Maui.Essentials.DeviceTests.exe
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '10.0.0-preview.3.25155.15' (x64)
.NET location: C:\hostedtoolcache\windows\dotnet

The following frameworks were found:
  9.0.0 at [C:\hostedtoolcache\windows\dotnet\shared\Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=10.0.0-preview.3.25155.15&arch=x64&rid=win-x64&os=win10
##[error]The Operation will be canceled. The next steps may not contain expected logs.
Terminate batch job (Y/N)?
```

--- a/eng/devices/windows.cake
+++ b/eng/devices/windows.cake
@@ -164,6 +164,7 @@ Task("buildOnly")
 	else
 	{
 		// Apply correct build properties for unpackaged builds
+		s.MSBuildSettings.Properties.Add("SelfContained", new List<string> { "True" });
 		s.MSBuildSettings.Properties.Add("WindowsPackageType", new List<string> { "None" });
 		s.MSBuildSettings.Properties.Add("ExtraDefineConstants", new List<string> { "UNPACKAGED" });
 	}

--- a/eng/pipelines/device-tests.yml
+++ b/eng/pipelines/device-tests.yml
@@ -120,14 +120,14 @@ stages:
         androidApiLevels: [ 33, 30, 29, 28, 27, 26, 25, 24, 23 ]
         iosVersions: [ 'simulator-18.0']
         catalystVersions: [ 'latest' ]
-        windowsVersions: ['packaged']
+        windowsVersions: ['packaged', 'unpackaged']
         provisionatorChannel: ${{ parameters.provisionatorChannel }}
         skipProvisioning: ${{ or(not(parameters.UseProvisionator), false) }}
       ${{ else }}:
         androidApiLevels: [ 33, 23 ]
         iosVersions: [ 'simulator-18.0' ]
         catalystVersions: [ 'latest' ]
-        windowsVersions: ['packaged']
+        windowsVersions: ['packaged', 'unpackaged']
         provisionatorChannel: ${{ parameters.provisionatorChannel }}
         skipProvisioning: ${{ not(parameters.UseProvisionator) }}
       projects:

